packages/lisk-transactions/src/transactions/base.ts

```diff
@@ -222,199 +226,62 @@ export abstract class BaseTransaction {
-		const errors: TransactionError[] = [];
-		// Check senderPublicKey
-		if (sender.publicKey !== this.senderPublicKey) {
-			errors.push(
-				new TransactionError(
-					'Invalid sender publicKey',
-					this.id,
-					'.senderPublicKey',
-				),
-			);
-		}
-
-		// Check senderId
-		if (
-			typeof sender.address === 'string' &&
-			sender.address.toUpperCase() !== this.senderId.toUpperCase()
-		) {
-			errors.push(
-				new TransactionError('Invalid sender address', this.id, '.senderId'),
-			);
-		}
-
-		// Check missing secondPublicKey on account
-		if (this.signSignature && !sender.secondPublicKey) {
-			errors.push(
-				new TransactionError('Sender does not have a secondPublicKey', this.id),
-			);
-		}
+	public verifyAgainstOtherTransactions(
+		transactions: ReadonlyArray<TransactionJSON>,
+	): TransactionResponse {
+		const errors = this.verifyAgainstTransactions(transactions);
 
-		// Balance verification
-		const { verified: balanceVerified, error: balanceError } = verifyBalance(
-			sender,
-			this.fee,
-		);
+		return createResponse(this.id, errors);
+	}
 
-		if (!balanceVerified && balanceError) {
-			errors.push(balanceError);
-		}
+	public apply(store: StateStore): TransactionResponse {
+		const errors = this._verify(store) as TransactionError[];
 
-		// Signature verifications
-
-		// Verify secondPublicKey
-		if (sender.secondPublicKey) {
-			// Check for missing signSignature
-			if (!this.signSignature) {
-				errors.push(
-					new TransactionError(
-						'Missing signSignature',
-						this.id,
-						'.signSignature',
-					),
-				);
-			} else {
-				const transactionBytes = Buffer.concat([
-					this.getBasicBytes(),
-					hexToBuffer(this.signature),
-				]);
-
-				const {
-					verified: signatureVerified,
-					error: verificationError,
-				} = verifySignature(
-					sender.secondPublicKey,
-					this.signSignature,
-					transactionBytes,
-					this.id,
-				);
-
-				if (!signatureVerified) {
-					errors.push(verificationError as TransactionError);
-				}
-			}
+		// Verify MultiSignature
+		const { errors: multiSigError } = this.processMultisignatures(store);
+		if (multiSigError) {
+			errors.push(...multiSigError);
 		}
-		// Verify multisignatures
 		if (
-			!(
-				sender.multisignatures &&
-				sender.multisignatures.length > 0 &&
-				sender.multimin
-			)
+			this._multisignatureStatus === MultisignatureStatus.PENDING &&
+			errors.length === 0
 		) {
-			this._multisignatureStatus = MultisignatureStatus.NONMULTISIGNATURE;
-
-			return {
-				id: this.id,
-				status: errors.length === 0 ? Status.OK : Status.FAIL,
-				errors,
-			};
-		}
-		this._multisignatureStatus = MultisignatureStatus.PENDING;
-		const {
-			status,
-			errors: multisignatureErrors,
-		} = this.processMultisignatures(sender);
-
-		if (status === Status.PENDING && errors.length === 0) {
 			return {
 				id: this.id,
-				status,
-				errors: multisignatureErrors,
+				status: Status.PENDING,
```

Comment:
In case of a pending transaction, the `apply` function should still update the account state and return the status as `pending`.